### PR TITLE
Clarify allocation and initial value of free memoy pointer.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

**Comment by axic** [comment]
The example updated in #4722 still needs to be updated.

**Comment by chriseth** [comment]
Ah sorry, I misread the example.

**Comment by chriseth** [comment]
Updated.

File: docs/assembly.rst [hunk 1]
```diff
@@ -609,13 +609,17 @@ first.
 
 Solidity manages memory in a very simple way: There is a "free memory pointer"
 at position ``0x40`` in memory. If you want to allocate memory, just use the memory
```
Comment (axic) [inline docs/assembly.rst:611]: This sentence is now misleading.

Comment (chriseth) [reply docs/assembly.rst:611]: It has always been...

**Comment by chriseth** [comment]
Updated.

**Review by axic — APPROVED** [review]

File: docs/assembly.rst [hunk 2]
```diff
@@ -609,12 +609,21 @@ first.
...
+starting from where this pointer points at and update it accordingly.
+There is no built-in mechanism to release or free allocated memory.
+Here is an assembly snippet that can be used for allocating memory::
+
+    function allocate(length) -> pos {
+      pos := mload(0x40)
+      mstore(0x40, add(pos, length))
+    }
 
 The first 64 bytes of memory can be used as "scratch space" for short-term
```
Comment (axic) [inline docs/assembly.rst:621]: If we want to be precise we would add a note here that this scratch space starts at `0x00`.

Likely this entire section should be rewritten in a better form. It shows that things have been added to it in iterations.

Comment (axic) [reply docs/assembly.rst:621]: Created an issue for it: #4728

**Pull request merged**